redesign/generator_builder.py

```diff
@@ -1,29 +1,48 @@
 """Contains generator factory class that picks generator based on its profile type."""
 
-from .generators import GraphGenerator, TabularGenerator, UnstructuredGenerator
 from dataprofiler import StructuredProfiler, UnstructuredProfiler
 
+from dataprofiler.profilers.graph_profiler import GraphProfiler
+
+from .generators import GraphGenerator, TabularGenerator, UnstructuredGenerator
+
+
 class Generator:
     """Generator class."""
 
     valid_data_types = {
-        StructuredProfiler: TabularGenerator, 
-        UnstructuredProfiler: UnstructuredGenerator
+        StructuredProfiler: TabularGenerator,
+        UnstructuredProfiler: UnstructuredGenerator,
+        GraphProfiler: GraphGenerator,
     }
 
     @classmethod
     def is_valid_data(cls, profile):
         return profile.__class__ in cls.valid_data_types
 
-    def __new__(cls, profile, config=None, options=None):
+    def __new__(cls, seed=None, config=None, *args, **kwargs):
         """Instantiate proper generator."""
         if config:
             try:
-                return config(profile, options)
+                return config(*args, **kwargs)
             except Exception as e:
-                print("Warning: profile doesn't match user setting. \
-                        Proceeding with automatic generator selection...")         
-                        
-        if cls.is_valid_data(profile):
-            generator = cls.valid_data_types[profile.__class__]
-            return generator(profile, options)
+                print(
+                    "Warning: profile doesn't match user setting. \
+                        Proceeding with automatic generator selection..."
+                )
```

Comment:
- this looks off to me because if it goes into this `try/except` but `return config()` has an exception , nothing really happens after that, no? 
- also, this scenario is not tested in the testing suite from what I'm seeing... should probably add a test case for this part of the logic

Reply:
If there is an invalid config, it'll print the error and then run through the other code. Should I make a test for a valid config?